common/src/main/scala/explore/components/graphql/LiveQueryRenderMod.scala

```diff
@@ -40,25 +38,21 @@ final case class LiveQueryRenderMod[S, D, A](
     Reusable.always(t => Message(error = true)(t.getMessage)),
   val onNewData:       IO[Unit] = IO.unit
 )(implicit
-  val F:               ConcurrentEffect[IO],
-  val timer:           Timer[IO],
-  val cs:              ContextShift[IO],
+  val F:               Async[IO],
+  val dispatcher:      Dispatcher[IO],
   val logger:          Logger[IO],
   val reuse:           Reusability[A],
   val client:          WebSocketClient[IO, S]
 ) extends ReactProps(LiveQueryRenderMod.component)
     with LiveQueryRenderMod.Props[IO, S, D, A]
 
 object LiveQueryRenderMod {
-  trait Props[F[_], S, D, A] extends Render.LiveQuery.Props[F, ViewF[F, *], S, D, A] {
-    implicit val timer: Timer[F]
-    implicit val cs: ContextShift[F]
-  }
+  trait Props[F[_], S, D, A] extends Render.LiveQuery.Props[F, ViewF[F, *], S, D, A]
 
   final case class State[F[_], S, D, A](
     queue:                   Queue[F, A],
     subscriptions:           NonEmptyList[GraphQLSubscription[F, _]],
-    cancelConnectionTracker: CancelToken[F],
+    cancelConnectionTracker: F[Unit],
```

Comment:
I thought the cancel token would be `() => Future[Unit]`

Reply:
That's for things that are running from outside of the effect system, started with `unsafe*Run`. In this case, everything is still happening within `F` since it's started with `f.start`. So the `CancelToken` is `fiber.cancel` (which is a `IO[Unit]`).

Reply:
Just to be clear: we could have done things this way before, and it would have been simpler from the start.